src/base/controller.hpp

```diff
@@ -334,16 +324,18 @@ void Controller::clear_parallelization() {
   parallel_state_update_ = 1;
 
   explicit_parallelization_ = false;
+  max_memory_mb_ = get_system_memory_mb() / 2;
 }
 
-void Controller::set_parallelization_experiments(const std::vector<Circuit>& circuits) {
+void Controller::set_parallelization_experiments(const std::vector<Circuit>& circuits,
+                                                 const Noise::NoiseModel& noise) {
 
   if (max_parallel_experiments_ <= 0)
     return;
   // if memory allows, execute experiments in parallel
   std::vector<size_t> required_memory_mb_list;
```

Comment:
just find out that we can set the initial size of the vector here, as we know it before hand: `circuits.size()`. This will avoid many allocations due to the `push_back`

Reply:
I didn't change any implementation of code not concerning the stored config/noise model in this PR, but if you notice some more simple changes (like this) I can add them

Reply:
Yeah, I know it's not in your changeset, just saw it while reviewing the code. If you can change this that would be great.